src/components/MetadataPanel/index.tsx

```diff
@@ -0,0 +1,148 @@
+import React, { useEffect, useMemo } from "react";
+import { Divider } from "antd";
+import classNames from "classnames";
+
+import { AgentMetadata, PositionRotation } from "../../constants/interfaces";
+import { FilledCaret } from "../Icons";
+import { AgentDisplayNode } from "../AgentTree";
+
+import styles from "./style.css";
+
+const MetadataLabels: Record<keyof AgentMetadata, JSX.Element> = {
```

Comment:
I think it would be good to move this within the functional component, it just confuses me up here